util/data_collection.py

```
@@ -0,0 +1,118 @@
+        try:
+            error_data = response.json()
+        except ValueError:
+            error_data = response.text
+
+        return {
+            "success": False,
+            "status_code": response.status_code,
+            "error": f"HTTP {response.status_code}: {response.reason}",
+            "data": error_data,
+            "headers": headers,
+        }
```

Ruff (0.14.11)

Consider moving this statement to an `else` block

(TRY300)